app/src/components/common/PinExtensionPopup/hooks/usePinExtensionPopup.ts

```diff
@@ -0,0 +1,20 @@
+import { useCallback, useState } from "react";
+import { useSelector } from "react-redux";
+import { getUserRulesCount } from "store/selectors";
+
+const IS_PIN_EXTENSION_POPUP_ACTIVE = "is_rq_pin_extension_popup_active";
+
+export const usePinExtensionPopup = () => {
+  const rulesCount = useSelector(getUserRulesCount);
+
+  const [isPinExtensionPopupActive, setIsPinExtensionPopupActive] = useState<boolean>(
+    JSON.parse(window.localStorage.getItem(IS_PIN_EXTENSION_POPUP_ACTIVE)) ?? rulesCount > 0
+  );
+
+  const closePinExtensionPopup = useCallback(() => {
+    setIsPinExtensionPopupActive(false);
+    window.localStorage.setItem(IS_PIN_EXTENSION_POPUP_ACTIVE, "false");
+  }, []);
```

Comment:
Why are we using `window.localStorage` instead of redux store?

Reply:
I think we should not put everything into redux store just for using localStorage via store.